src/server/editorServices.ts

```diff
@@ -783,6 +789,10 @@ namespace ts.server {
                     this.openFileRoots = openFileRoots;
                     this.openFileRoots.push(info);
                 }
+
+                if (fileExtensionIs(info.fileName, ".js") || fileExtensionIs(info.fileName, ".jsx")) {

```

Comment:
This line is only for inferred project that doesn't have a config file, namely loose files. So currently I am not inferring typings for loose `.ts` files. For projects with config file, one can set typing options in the config file for either `tsconfig.json` or `jsconfig.json`, though all the inferring still only happen on `.js` files